src/libs/sdlcd/linux/SDL_syscdrom.c

```diff
@@ -0,0 +1,574 @@
+		return(-1);
+	cdfd = open(drive, (O_RDONLY|O_NONBLOCK), 0);
+	if ( stat(drive, stbuf) < 0 ) {
+		if ( cdfd >= 0 )
+			close(cdfd);
+		return(-1);
+    }
+
+	/* If it does exist, verify that it's an available CD-ROM */
+	is_cd = 0;
+	if ( S_ISCHR(stbuf->st_mode) || S_ISBLK(stbuf->st_mode) ) {
+		if ( cdfd >= 0 ) {
+			info.cdsc_format = CDROM_MSF;
+			/* Under Linux, EIO occurs when a disk is not present.
+			 */
+			if ( (ioctl(cdfd, CDROMSUBCHNL, &info) == 0) ||
+						ERRNO_TRAYEMPTY(errno) ) {
+				is_cd = 1;
+			}
+		}
+#ifdef USE_MNTENT
+		/* Even if we can't read it, it might be mounted */
+		else if ( mnttype && (SDL_strcmp(mnttype, MNTTYPE_CDROM) == 0) ) {
+			is_cd = 1;
+		}
+#endif
+	}
+	if ( cdfd >= 0 )
+		close(cdfd);
+	return(is_cd);
+}
+
+/* Add a CD-ROM drive to our list of valid drives */
+static void AddDrive(char *drive, struct stat *stbuf)
+{
+	int i;
+
+	if ( SDL_numcds < MAX_DRIVES ) {
+		/* Check to make sure it's not already in our list.
+	 	   This can happen when we see a drive via symbolic link.
+		 */
+		for ( i=0; i<SDL_numcds; ++i ) {
+			if ( stbuf->st_rdev == SDL_cdmode[i] ) {
+#ifdef DEBUG_CDROM
+  fprintf(stderr, "Duplicate drive detected: %s == %s\n", drive, SDL_cdlist[i]);
+#endif
+				return;
+			}
+		}
+
+		/* Add this drive to our list */
+		i = SDL_numcds;
+		SDL_cdlist[i] = SDL_strdup(drive);
+		if ( SDL_cdlist[i] == NULL ) {
+			SDL_OutOfMemory();
+			return;
+		}
+		SDL_cdmode[i] = stbuf->st_rdev;
+		++SDL_numcds;
+#ifdef DEBUG_CDROM
+  fprintf(stderr, "Added CD-ROM drive: %s\n", drive);
+#endif
+	}
+}
+
+#ifdef USE_MNTENT
+#define STRMAXLEN 300
+static void CheckMounts(const char *mtab)
+{
+	FILE *mntfp;
+	struct mntent *mntent;
+	struct stat stbuf;
+
+	mntfp = setmntent(mtab, "r");
+	if ( mntfp != NULL ) {
+		char *tmp;
+		size_t mnt_type_len;
+		size_t mnt_dev_len;
+		char mnt_type_str[STRMAXLEN+1];
+		char mnt_dev_str[STRMAXLEN+1];
+		char *mnt_type = mnt_type_str;
+		char *mnt_dev = mnt_dev_str;
+
+		while ( (mntent=getmntent(mntfp)) != NULL ) {
+			*mnt_type = '\0';
+			*mnt_dev = '\0';
+			mnt_type_len = SDL_strlen(mntent->mnt_type) + 1;
+			if (mnt_type_len>STRMAXLEN)
+				strncpy(mnt_type_str, mntent->mnt_type, STRMAXLEN);
+			else
+				strcpy(mnt_type_str, mntent->mnt_type);
+			mnt_type_str[STRMAXLEN] = '\0';
+
+			mnt_dev_len = SDL_strlen(mntent->mnt_fsname) + 1;
+			if (mnt_dev_len>STRMAXLEN)
+				strncpy(mnt_dev_str, mntent->mnt_fsname, STRMAXLEN);
+			else
+				strcpy(mnt_dev_str, mntent->mnt_fsname);
+			mnt_dev_str[STRMAXLEN] = '\0';
+
+			SDL_strlcpy(mnt_type, mntent->mnt_type, mnt_type_len);
+			SDL_strlcpy(mnt_dev, mntent->mnt_fsname, mnt_dev_len);
+
+			/* Handle "supermount" filesystem mounts */
+			if ( SDL_strcmp(mnt_type, MNTTYPE_SUPER) == 0 ) {
+				tmp = SDL_strstr(mntent->mnt_opts, "fs=");
+				if ( tmp ) {
+					strcpy(mnt_type, strlen(tmp) + SDL_strlen("fs=") <= STRMAXLEN ? tmp + SDL_strlen("fs=") : "");
+					if ( *mnt_type ) {
+						tmp = SDL_strchr(mnt_type, ',');
```

Comment:
Unbounded C-string copy defect caught by Coverity:

![2022-02-14_14-50](https://user-images.githubusercontent.com/1557255/153960276-8c5dc520-35d1-49d5-b276-bb10d1fc507b.png)

Reply:
Maybe the latest code fixes it.